lib/kuniri/state_machine/OO_structured_fsm/constructor_state.rb

```diff
@@ -1,66 +1,25 @@
 require_relative 'oo_structured_state'
+require_relative 'method_state'
 
 module StateMachine
 
   module OOStructuredFSM
 
     # Class responsible for handling constructor state.
-    class ConstructorState < OOStructuredState
+    class ConstructorState < MethodState
 
-      @language
+      CLASS_TYPE = "constructor"
```

Comment:
I like this idea! However, I believe that isn't a good idea to set a string directly to the attribute as you did. I recommend you to create some constants in token_state_machine.rb and just use it here. 
Something like this:

``CLASS_TYPE ||= StateMachice::CLASS_TYPE``